tests/test_contact.py

```
@@ -0,0 +1,435 @@
+    cursor = conn.cursor()
+    cursor.execute("SELECT email FROM contact_submissions WHERE id = ?", (submission_id,))
+    email = cursor.fetchone()[0]
+    conn.close()
+    
+    assert email == "john@example.com"
+
+
+def test_rate_limiting(client, monkeypatch):
+    """Test rate limiting prevents excessive submissions"""
+    # Mock IP address
+    def mock_get_client_ip(request):
```

Ruff (0.14.10)

Unused function argument: `request`

(ARG001)